fuse/daemon.cpp

```diff
@@ -1405,10 +1420,12 @@ static void handle_exit(int sig)
 		int status;
 		do waitpid(pid, &status, 0);
 		while (WIFSTOPPED(status));
+		// Attempts numbered counting from 1:
+		fprintf(stderr, "Unable to umount on attempt %d\n", exit_attempts);
 	}
 
-	if (exit_attempts == 3) {
-		fprintf(stderr, "Unable to cleanly exit after 4 unmount attempts\n");
+	if (exit_attempts == QUIT_RETRY_ATTEMPTS) {
+		fprintf(stderr, "Too many umount attempts; unable to exit cleanly. Leaving a broken mount point behind.\n");
```

Comment:
What causes unmount failures?

Reply:
Anything lingering in the .fuse/ folder will cause the umount call too fail. Unfortunately, there is no way to do the umount synchronously within the FUSE loop, so it's inherently failure-prone and racey. If the daemon calls `exit(1)` before the umount succeeds, you get one of those weird dangling mount points that report '????' for their attributes and you have to manually `fusermount -u`. That's why the daemon retries pretty hard not to do an exit.